doc/source/dev/toolchain.rst

```diff
@@ -252,18 +253,54 @@ the bottleneck for C++ support is therefore the oldest supported GCC version,
-that only comes with GCC 6.3 [18]_.
-
-C++17 _language_ support will require GCC >= 7 (released May 2017). As of the
-end of 2021, support for the entirety of the C++17 standard library has not yet
-been completed across all compilers; similarly, support for C++20 and C++23
-is still under heavy development. [17]_
+At the end of 2021 (with the final removal of ``manylinux1`` wheels), the
+minimal requirement of GCC moved to 6.3, which has full C++14 support [17]_.
+This corresponded to the lowest-present GCC version in relevant manylinux
+versions, though this was still considering the Debian-based "outlier"
+``manylinux_2_24``, which - in contrast to previous manylinux images based on
+RHEL-derivative CentOS that could benefit from the ABI-compatible GCC backports
+in the "RHEL Dev Toolset" - was stuck with GCC 6.3. That image failed to take
+off not least due to those outdated compilers [18]_ and reached its EOL in
+mid-2022 [19]_. For different reasons, ``manylinux2010`` also reached its EOL
+around the same time [20]_.
+
+The remaining images ``manylinux2014`` and ``manylinux_2_28`` currently support
+GCC 10 and 11, respectively. The latter will continue to receive updates as new
+GCC versions become available as backports, but the former will likely not
+change since the CentOS project is not responsive anymore about publishing
+aarch64 backports of GCC 11 [21]_.
+
+This leaves all the main platforms and their compilers with comparatively
+recent versions. However, SciPy has historically also endeavored to support
+less common platforms as well - if not with binary artefacts (i.e. wheels),
+then at least by remaining compilable from source - which includes for example
+AIX, Alpine Linux and FreeBSD.
+
+For AIX 7.1 & 7.2 the default compiler is GCC 8 (AIX 6.1 had its EOL in 2017),
+but GCC 10 is installable (side-by-side) [22]_.
+The oldest currently-supported Alpine Linux release is 3.12 [23]_, and already
+comes with GCC 10.
+For FreeBSD, the oldest currently-supported 12.x release [24]_ comes with
+LLVM 10 (and GCC 10 is available as a freebsd-port [25]_).
+
+Finally there is the question of which machines are widely used by people
+needing to compile SciPy from source for other reasons (e.g. SciPy developers,
+or people wanting to compile for themselves or performance reasons). The oldest
+relevant distributions (without RHEL-style backports) are Ubuntu 18.04 LTS
+(which has GCC 7 but also has a backport of GCC 8; Ubuntu 20.04 LTS has GCC 9)
+and Debian Buster (with GCC 8; Bullseye has GCC 10).
+This is the weakest restriction for determining the lower bounds of compiler
+support (power users and developers can be expected to keep their systems at
+least somewhat up-to-date, or use backports where available), and gradually
+becomes less important as usage numbers of old distributions dwindle.
+
+All the lowest-supported compiler versions (GCC 8, LLVM X, VS2019 with vc142)
```

Comment:
As explained in the OP, this is a draft & a suggestion. I'm also not sure we want to promise a lower bound for LLVM at all.